sdk/ml/azure-ai-ml/azure/ai/ml/_local_endpoints/mdc_config_resolver.py

```diff
@@ -0,0 +1,81 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import json
+import os.path
+from pathlib import Path
+
+from azure.ai.ml.entities._deployment.data_collector import DataCollector
+
+
+class MdcConfigResolver(object):
+
+    """Represents the contents of mdc config and handles writing the mdc configuration to User's system.
+
+    :param data_collector: model data collector entity
+    :type data_collector: DataCollector
+    """
+
+    def __init__(
+        self,
+        data_collector: DataCollector,
+    ):
+        self.environment_variables = None
```

Comment:
nit: would it make sense to initialize `environment_variables` and `volumes` as empty dicts instead of `None`?

Reply:
thanks, fixed

Reply:
I've done that but it still doesn't work